src/node_nanomsg.cc

```diff
@@ -205,23 +205,21 @@ NAN_METHOD(Shutdown) {
 
 
 NAN_METHOD(Send) {
-    NanScope();
-
-    int s = args[0]->Uint32Value();
-
-    if(!node::Buffer::HasInstance(args[1])) {
-        return NanThrowError("second Argument must be a Buffer.");
-    }
+  NanScope();
 
-    Local<Object> obj = args[1]->ToObject();
-    char* odata = node::Buffer::Data(obj);
-    size_t odata_len = node::Buffer::Length(obj);
-    int flags = args[2]->Uint32Value();
+  int flags = args[2].integer;
+  std::string *input;
 
-    // Invoke nanomsg function.
-    int ret = nn_send(s, odata, odata_len, flags);
+  if (node::Buffer::HasInstance(args[1]->ToObject())) {
+    v8::Handle<v8::Object> object = args[1]->ToObject();
+    const char *data = node::Buffer::Data(object);
+    input = new std::string(data, node::Buffer::Length(object));
+  } else {
+    utf8 str (args[1]->ToString());
+    input = new std::string(*str);
```

Comment:
eh true, for all i know that `str` could be billowing out onto the heap.

I must have wanted `nn_send()` to go down outside the condition i guess

but now that you mention it we need to fix those two things.. let's just confirm the performance first, I don't want to make this change if it's faster to send the binding a buffer and use node javascript to do the conversion (which i kinda doubt, but who knows)

so i'll run some benchmarks on small and large message envelopes later tonight and confirm then